controller/k8s/pods.go

```diff
@@ -26,7 +26,7 @@ type podIndex struct {
 	stopCh    chan struct{}
 }
 
-func NewPodIndex(clientset *kubernetes.Clientset, index cache.IndexFunc) (PodIndex, error) {
+func NewPodIndex(clientset kubernetes.Interface, index cache.IndexFunc) (PodIndex, error) {
```

Comment:
is this necessary to support a refactor or just housekeeping?

Reply:
This particular change was not necessary, but an analogous one was required in `replicasets.go` to enable pass in a fake k8s client, so thought err'd on the side of consistency.